sdk/resource/resource.go

```diff
@@ -72,7 +91,17 @@ func NewWithAttributes(attrs ...attribute.KeyValue) *Resource {
 		return &emptyResource
 	}
 
-	return &Resource{s} //nolint
+	return &Resource{attrs: s} //nolint
+}
+
+// Is there a better way that does not require introducing a NewWithAttributesAndSchemaURL
+// function? I tried adding a schemaURL to NewWithAttributes but there are a lot of places
+// where the schema url should be "" so it seemed somewhat ugly to have all those callsites
+// with empty strings as first parameter.
+func NewWithAttributesAndSchemaURL(schemaURL string, attrs ...attribute.KeyValue) *Resource {
```

Comment:
Maybe these should all just use `New()` with appropriate `Option`s?

```go
res, err := resource.New(ctx,
	WithSchemaURL(schemaURL),
	WithAttributes(attrs...))
```

Reply:
The downside is that `resource.New` returns a tuple (Resource,error) and it will require to touch a lot of code to handle the returned error, but kind of is pointless since there is no way for the call to actually fail.
I would slightly prefer this to be a separate function since it appears to be a common use case. Maybe we can find a better/shorter name for `NewWithAttributesAndSchemaURL`?

Reply:
I pushed a new commit which renames NewWithAttributes to NewSchemaless and makes schema url mandatory for NewWithAttributes.

We want to encourage using schema url and make it a concious choice to have a resource without schema. These names seem to be better suited for that goal.

What do you think?

Reply:
Obviously this is a breaking change for existing code. I assume we are OK with breaking it since we are not yet stable. I'll revert if the assumption is incorrect.

Reply:
I think that is okay.  This seems like a good enough reason to change the API and it aligns the API better with what we want to support long-term.  @MrAlias thoughts?

Reply:
I'm okay with breaking changes, but want to make sure the stuff we are introducing is going to be around for the long haul. I would rather "rip the bandage off" now than have to support deprecated functions right when we do a release.

Reply:
I think what we have can stay long term.